zetaclient/keys.go

```diff
@@ -164,3 +182,28 @@ func (k *Keys) GetPubKeySet() (common.PubKeySet, error) {
 	pubkeySet.Secp256k1 = pubkey
 	return pubkeySet, nil
 }
+
+// GetHotkeyPassword returns the password to be used
+// returns empty if no password is needed
+func (k *Keys) GetHotkeyPassword() (string, error) {
+	if k.GetKeybase().Backend() == ckeys.BackendFile {
+		return getHotkeyPassword()
+	}
+	return "", nil
+}
+
+// getHotkeyPassword retrieves the HOTKEY_PASSWORD environment variable
+// and returns an error if it's not defined or shorter than 8 characters.
+func getHotkeyPassword() (string, error) {
+	password := os.Getenv(HotkeyPasswordEnvVar)
```

Comment:
As further improvement, i think storing the password in process memory is better than storing it in environment variable. The reason is that envivronment variable are in known places in a process (or maybe just the control block of a process?). I.e., I can on linux simply find the all environment variables in a running process: by
`$ cat /proc/2328594/environ `

If the password is stored in a regular process variable it will be much harder to find even if a shell is obtained by an attacker on that node that runs `zetaclientd`. 

This is not blocking this PR though so feel free to comment and resolve.

Reply:
It could also be added as an additional option as well.
The `KeyringBackend` options we define in ZetaClient don't need to match exactly the `KeyringBackend` options of Cosmos (maybe should rename the option then).
We could have for example `KeyringBackendFilePasswordEnv` and `KeyringBackendFilePasswordMemory`.
I may think env var option can be convenient for teams that want to run a ZetaClient with a fullly Infrastructure as Code design.

Reply:
I agree that ENV should certainly be an option, but when we publish documentation we really should educate folks on the safety of their keys. Perhaps we should even go as far as to present a ranked list of options or flowchart.

I am still encountering people who think their keys/cryptocurrency is safe on a hardware wallet and therefore do not need to protect/worry about the mnemonic's security.

Reply:
https://github.com/zeta-chain/node/issues/1368